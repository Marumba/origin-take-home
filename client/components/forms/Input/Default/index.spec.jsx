import React from 'react';
import { css } from '@emotion/react';
import userEvent from '@testing-library/user-event';

import { render, screen } from '~/helpers/testing.customRtl';

import InputDefault from './';

describe('Input Text', () => {
	const testCss = () => css`
		background: red;
	`;
	it('should render with the required prop values', () => {
		const { container } = render(<InputDefault id="ad" name="test" label="test" />);
		expect(screen.getByLabelText('test')).toBeInTheDocument();
		expect(container.firstChild).toMatchSnapshot();
	});

	it('should display the label prop value', () => {
		render(<InputDefault id="ad" name="test" label="Test" />);
		expect(screen.getByLabelText('Test')).toBeInTheDocument();
	});

	it('should incorporate de given css', () => {
		const { container } = render(<InputDefault id="ad" name="test" label="Test" customCss={testCss} />);
		expect(container.firstChild).toHaveStyleRule('background', 'red');
	});

	it('should be a controlled input if prop onChange is available', () => {
		render(<InputDefault id="ad" name="test" label="Test" value="controlled" onChange={f => f} />);

		userEvent.type(screen.getByRole('textbox'), 'No changes');

		expect(screen.queryByDisplayValue('No changes')).toBeNull();
		expect(screen.getByDisplayValue('controlled')).toBeInTheDocument();
	});

	it('should be an uncontrolled input by default', () => {
		render(<InputDefault id="ad" name="test" label="Test" />);

		userEvent.type(screen.getByRole('textbox'), 'New Value');

		expect(screen.getByDisplayValue('New Value')).toBeInTheDocument();
	});
});
