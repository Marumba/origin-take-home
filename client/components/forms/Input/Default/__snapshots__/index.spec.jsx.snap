// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Text should render with the required prop values 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-2 {
  font-family: 'Work Sans',sans-serif;
  font-weight: 400;
  font-size: 12px;
  line-height: 150%;
  margin-bottom: 4px;
  color: #1E2A32;
}

@media screen and (min-width: 992px) {
  .emotion-2 {
    font-size: 14px;
  }
}

.emotion-4 {
  border: 1px solid #E9EEF2;
  border-radius: 4px;
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 56px;
  position: relative;
  padding: 0 14px;
}

.emotion-4 svg+input {
  padding-left: 7px;
}

.emotion-6 {
  font-family: 'Rubik',sans-serif;
  font-weight: 500;
  font-size: 20px;
  line-height: 120%;
  color: #4D6475;
  padding: 14px 0;
  width: 100%;
}

@media screen and (min-width: 992px) {
  .emotion-6 {
    font-size: 24px;
  }
}

.emotion-6.highEmphasis {
  color: #1E2A32;
}

<div
  class="emotion-0 emotion-1"
>
  <label
    class="emotion-2 emotion-3"
    for="ad"
  >
    test
  </label>
  <div
    class="emotion-4 emotion-5"
  >
    <input
      class="emotion-6 emotion-7"
      id="ad"
      name="test"
      type="text"
      value=""
    />
  </div>
</div>
`;
